datalake/src/test/java/com/sequenceiq/datalake/service/validation/cloudstorage/CloudStorageValidatorTest.java

```diff
@@ -92,9 +97,51 @@ public void validateEnvironmentRequestCloudStorageValidation() {
         when(entitlementService.cloudStorageValidationEnabled(any())).thenReturn(true);
-        when(cloudProviderServicesV4Endopint.validateObjectStorage(any())).thenReturn(new ObjectStorageValidateResponse());
+        when(cloudProviderServicesV4Endpoint.validateObjectStorage(any())).thenReturn(new ObjectStorageValidateResponse());
         ValidationResultBuilder validationResultBuilder = new ValidationResultBuilder();
         ThreadBasedUserCrnProvider.doAs(USER_CRN, () -> underTest.validate(new CloudStorageRequest(), environment, validationResultBuilder));
         assertFalse(validationResultBuilder.build().hasError());
     }
+
+    @Test
+    public void validateBackupLocation() {
+        when(environment.getCredential()).thenReturn(new CredentialResponse());
+        when(secretService.getByResponse(any())).thenReturn("secret");
+        when(regionAwareInternalCrnGenerator.getInternalCrnForServiceAsString()).thenReturn("crn");
+        when(regionAwareInternalCrnGeneratorFactory.iam()).thenReturn(regionAwareInternalCrnGenerator);
+        when(credentialToCloudCredentialConverter.convert(any())).thenReturn(
+                new CloudCredential("id", "name", Map.of("secretKey", "thisshouldnotappearinlog"), "acc", false));
+
+        when(cloudProviderServicesV4Endpoint.validateObjectStorage(any())).thenReturn(new ObjectStorageValidateResponse());
+        ValidationResultBuilder validationResultBuilder = new ValidationResultBuilder();
+        ThreadBasedUserCrnProvider.doAs(USER_CRN, () -> underTest.validateBackupLocation(new CloudStorageRequest(), environment, validationResultBuilder));
+        assertFalse(validationResultBuilder.build().hasError());
+    }
+
+    @Test
+    public void validateBackupLocationOnError() {
+        ArgumentCaptor<ObjectStorageValidateRequest> captor = ArgumentCaptor.forClass(ObjectStorageValidateRequest.class);
+        when(environment.getCredential()).thenReturn(new CredentialResponse());
+        when(environment.getCloudPlatform()).thenReturn("AWS");
+        when(environment.getBackupLocation()).thenReturn("/location/to/backup");
+        when(regionAwareInternalCrnGenerator.getInternalCrnForServiceAsString()).thenReturn("crn");
+        when(regionAwareInternalCrnGeneratorFactory.iam()).thenReturn(regionAwareInternalCrnGenerator);
+        when(credentialToCloudCredentialConverter.convert(any())).thenReturn(
+                new CloudCredential("id", "name", Map.of("secretKey", "thisshouldnotappearinlog"), "acc", false));
+
+        ObjectStorageValidateResponse objectStorageValidateResponse = new ObjectStorageValidateResponse();
+        objectStorageValidateResponse.setStatus(ResponseStatus.ERROR);
+        objectStorageValidateResponse.setError("dummy failure");
+        when(cloudProviderServicesV4Endpoint.validateObjectStorage(captor.capture())).thenReturn(objectStorageValidateResponse);
```

Comment:
The `captor.capture()` invocation must appear within a `verify()` check (see [javadoc](https://javadoc.io/static/org.mockito/mockito-core/4.6.1/org/mockito/ArgumentCaptor.html#capture())). It is okay to have a wildcard arg matcher here in the `when()` stubbing, so e.g. `any()` or preferably `any(ObjectStorageValidateRequest.class)`.